contracts/FeeProvider/FeeCollector.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./FeeDealProvider.sol";
+import "./FeeLockProvider.sol";
+import "./FeeTimedProvider.sol";
+
+contract FeeCollector is IERC721Receiver, IFeeCollector {
+    uint256 public fee; // 1e18 = 100%
+    address public immutable feeCollector;
+    ILockDealNFT public immutable lockDealNFT;
+    IProvider public immutable feeDealProvider;
+    IProvider public immutable feeLockProvider;
+    IProvider public immutable feeTimedProvider;
```

Comment:
Let's save it in a mapping, like an approved contract. 
will need to add refund and bundle later.